### ISSUE TITLE
invalid apple id token: 'aud' must be a string containing the client_id v10 react-native

### ISSUE THREAD

**Issue by kevin-lii** [realm/realm-js#3122]
## Goals

Use Apple Sign In to log into the app

## Expected Results

Successful Sign in.

## Actual Results
I got an error:
`invalid apple id token: 'aud' must be a string containing the client_id`
## Steps to Reproduce
use invertase/react-native-apple-authentication to get the identity_token and pass the token into `Credentials.apple()`

## Version of Realm and Tooling

- Realm JS SDK Version: 10.0.0-beta.9
- React Native: 0.62.2

**Comment by kneth** [comment]
You do mean `identityToken`, right?

How closely do you follow https://github.com/invertase/react-native-apple-authentication/tree/master/example?

**Comment by jasontlouro** [comment]
Hey @kneth, is Realm compatible with `@invertase/react-native-apple-authentication`? I'm having the following issue:

- The setup docs for the above repo show that different Client IDs are used for Android and iOS versions of Apple login. On iOS, the Client ID that gets sent to Realm after successful login is the App ID - in our case `com.campfirewriting.campfiremobile`. On Android, the Client ID that gets sent to Realm is the Service ID we had to generate according to the `react-native-apple-authentication` docs [here](https://github.com/invertase/react-native-apple-authentication/blob/main/docs/ANDROID_EXTRA.md), in our case `com.campfirewriting.campfiremobile-appleweb`. These _must_ be distinct because Apple won't let you have an App ID and a Service ID with the same identifier.
- Therefore, when I have `com.campfirewriting.campfiremobile` set as the accepted Client ID in the Apple authentication provider on Realm, Sign in with Apple works on iOS but not Android, and vice versa when I set it to `com.campfirewriting.campfiremobile-appleweb`. 

Am I missing something here? If not, would it be possible to add the option for a comma-delimited list of acceptable Client IDs in the Apple authentication provider UI, as you did for `aud` in Custom JWT? As it stands, I have no way to support Sign in with Apple on both iOS and Android devices. I believe the problem only gets worse when we later try to add support for it to our website. Comma-delimited acceptable Client IDs would resolve the issue, I think.

Will also tag @kevin-lii in case he found a solution to this at the time. Thanks!
